thyrel-api/thyrel-api/Controllers/ElementController.cs

```diff
@@ -29,18 +29,20 @@ public ElementController(IWebsocketHandler websocketHandler, HolyDrawDbContext c
         public async Task<ActionResult<Element>> Finish(int id, [FromBody] ElementBody body)
         {
             var elementDataProvider = new ElementDataProvider(_context);
+            var sessionDataProvider = new SessionDataProvider(_context);
             var element = await elementDataProvider.GetElement(id);
-            var session = await new SessionDataProvider(_context).GetSessionById(element.SessionId);
-            await elementDataProvider.HandleFinish(id, true);
-
-            if (element.Type == ElementType.Sentence)
-            {
-                await elementDataProvider.SetSentence(id, body.Text);
-            }
-            else
-            {
-                await elementDataProvider.SetDrawing(id, body.DrawingId ?? 0);
-            }
+            var session = await sessionDataProvider.GetSessionById(element.SessionId);
+            
+            var finishState = await elementDataProvider.HandleFinish(id);
+            if (finishState.FinishAt != null)

```

Comment:
En fait la chose qui me chagrine un peu c'est qu'a chaque fois que le player va click sur "Finish" -> "Not finish" on va appeler l'endpoint et faire des opérations  dans la DB
je sais que suivant notre logique c'est tout à fait normal

Reply:
tu as raison Luca, il y a un problème ici.
A terme je pense qu'on vas rajouté une colonne dans Session car la on vas count le nombre de player tous le temps, compter le nombre d'element est quand-à-lui impossible à changer